javascript/javascript1/week3/main.js

```diff
@@ -0,0 +1,215 @@
+
+const notes = [];
+
+function saveNote(content, id) {
+  notes.push({ content: content, id: id });
+}
+
+saveNote("Pick up groceries", 1);
+saveNote("Do laundry", 2);
+saveNote("Kobe ind fra Netto", 3);
+saveNote("Wash the dishes", 4);
+
+console.log(notes);
+
+function getNote(id) {
+  for (let i = 0; i < notes.length; i++) {
+    if (id === notes[i].id) {
+      return notes[i];
+    } else {
+      return "You entered wrong id";
+    }
+  }
+}
+
+const firstNote = getNote(1);
+console.log(firstNote);
+
+function logOutNotesFormatted() {
+  for (let i = 0; i < notes.length; i++) {
+    console.log(
+      `The note with id: ${notes[i].id}, has the following note text: ${notes[i].content} `
+    );
+  }
+}
+logOutNotesFormatted();
+
+// My idea to make the app better
+function missionDone(id) {
+  for (let i = 0; i < notes.length; i++) {
+    if (id === notes[i].id) {
+      console.log(
+        `The mission \"${notes[i].content}\" is Done and hase been deleted from your notes`
+      );
+      notes.splice(i, 1);
+    }
+  }
+}
+
+missionDone(2);
+logOutNotesFormatted();
+console.log("#################");
+
+const activities = [];
+function addActivity(date, activity, duration) {
+  activities.push({ date: date, activity: activity, duration: duration });
+}
+addActivity("21/7/21", "Youtube", 10);
+addActivity("21/7/21", "Facebook", 8);
+addActivity("21/7/21", "Instagram", 12);
+addActivity("21/7/21", "WhatsApp", 14);
+addActivity("22/7/21", "Youtube", 15);
+addActivity("22/7/21", "Facebook", 7);
+addActivity("22/7/21", "Instagram", 21);
+addActivity("22/7/21", "WhatsApp", 7);
+console.log(activities);
+
+function showStatus(activities) {
+  let totalduration = 0;
+  const usageLimit = 90;
+  if (activities.length == 0) {
+    console.log("Add some activities before calling showStatus");
+  } else {
+    for (let i = 0; i < activities.length; i++) {
+      totalduration += activities[i].duration;
+    }
+    console.log(
+      `You have added ${activities.length} activities. They amount to ${totalduration} Minutes of usage `
+    );
+    totalduration > usageLimit
+      ? console.log("No more social media for you")
+      : console.log(
+          `you can still use social media for ${
+            usageLimit - totalduration
+          } minutes`
+        );
+  }
+}
+showStatus(activities);
+
+console.log("@@@@@@@@@");
+function showStatusImprovedVersion(activities, date) {
+  let totalduration = 0;
+  let specificDateActivities = 0;
+  const usageLimit = 45;
+  if (activities.length == 0) {
+    console.log("Add some activities before calling showStatus");
+  } else {
+    for (let i = 0; i < activities.length; i++) {
+      if (activities[i].date === date) {
+        totalduration += activities[i].duration;
+        specificDateActivities++;
+      }
+    }
+    console.log(
+      `You have added ${specificDateActivities} activities. They amount to ${totalduration} Minutes of usage `
+    );
+    totalduration > usageLimit
```

Comment:
Ternary operator works just great but it's a better practice to keep them either on one line or use if statement. You can read more here https://github.com/airbnb/javascript#comparison--nested-ternaries